src/plugins/RepairMan.cpp

```diff
@@ -2,15 +2,48 @@
 //
 // Copyright (c) 2017-2020 Alexander Kurbatov
 
+#include "Historican.h"
+#include "Hub.h"
 #include "RepairMan.h"
 #include "core/API.h"
 #include "core/Helpers.h"
 
+#include <sc2api/sc2_unit_filters.h>
+
+namespace {
+Historican gHistory("repairman");
+
+}  // namespace
+
 void RepairMan::OnStep(Builder*) {
-    if (gAPI->observer().GetCurrentRace() != sc2::Race::Terran)
+    if (gHub->GetCurrentRace() != sc2::Race::Terran || m_damaged_buildings.empty())
```

Comment:
This `m_damaged_buildings.empty()` check is not needed as below we have another one right after std::remove_if`. If removal done in one call, it'll be safe to check the list for emptiness later.